torch/_inductor/lowering.py

```diff
@@ -1015,7 +1015,9 @@ def inner_fn(idx):
 
             # if we're concatting [4], [2]
             # when we index the second tensor for 5 we want to index 5 - 4
-            idx_load[dim] -= inputs_ranges[i][0]
+            # Use Identity to prevent expansion of index * stride to keep expression
```

Comment:
So the root cause of the issue is

`-inputs_range[i][0] * stride` get underflow?